core/src/main/java/feign/Capability.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2012-2020 The Feign Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package feign;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.List;
+import feign.Logger.Level;
+import feign.Request.Options;
+import feign.codec.Decoder;
+import feign.codec.Encoder;
+
+public interface Capability {
```

Comment:
I don't really know what this component is meant to to do?  Can we add some documentation here?